### PR TITLE
Fix scatter_op python API 

### LINKED ISSUE #12725: fluid下scatter操作对输入参数要求有问题
a，labels，diff的dtype分别为float32, int64, float32
调用fluid下`layers.scatter(x=a,ids=labels,updates=diff)`，提示错误`operator scatter must input same dtype. VarType.FP32 vs VarType.INT64`

后来跟进代码里面，发现按照这个check逻辑，scatter的三个输入的dtype必须要求一致才行。
image

按照scatter的用法说明，ids为对应索引index，不应该是float类型吧？请问下这是什么原因呢？

### PR DESCRIPTION
**Pull request by ktlichkid**
and remove inconsistency between implementation and documentation

Resolved #12725

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5036,6 +5037,45 @@ def gather(input, index):
     return out
 
 
+def scatter(input, index, updates):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]:
```
def scatter(input, index, updates, name=None):
```

Please also add unit tests in https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/unittests/test_layers.py

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5036,6 +5037,45 @@ def gather(input, index):
...
+    axis.
+
+    .. math::
+
+        Out = X
+        Out[Ids] = Updates
+
+    Args:
+        input: The source input with rank>=1.
+        index: The index input with rank=1.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: Please tell user the type for index, int32_t or int64_t

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -5036,6 +5037,45 @@ def gather(input, index):
...
+    Output is obtained by updating the input on selected indices on the first
+    axis.
+
+    .. math::
+
+        Out = X
+        Out[Ids] = Updates
+
+    Args:
+        input: The source input with rank>=1.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: input (Variable): xxx

the same as below.

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (ktlichkid) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (ktlichkid) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (ktlichkid) [reply python/paddle/fluid/layers/nn.py]: Done

**Review by qingqing01 — DISMISSED** [review]

**Review by panyx0718 — COMMENTED** [review]
Is the API design following some other framework?

**Review by panyx0718 — DISMISSED** [review]
approve API

**Review by qingqing01 — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**